### PR TITLE
[HAProxy] Fix source.ip condition in append processors

### PR DESCRIPTION
**Pull request by Bearloggs**
Bug

## Proposed commit message

* Minor bug fix to source.ip conditions in append processors to avoid following error :  
```
cannot access method/field [ip] from a null def reference
```

## Checklist

- [X] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [X] I have verified that all data streams collect metrics or logs.
- [X] I have added an entry to my package's `changelog.yml` file.
- [X] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).
- [X] I have verified that any added dashboard complies with Kibana's [Dashboard good practices](https://docs.elastic.dev/ux-guidelines/data-viz/dashboard-good-practices)

### CODE REVIEW

File: packages/haproxy/data_stream/log/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -157,11 +157,11 @@ processors:
   - append:
       field: event.category
       value: network
-      if: "ctx?.source.ip != null && ctx?.destination?.ip != null"
+      if: "ctx?.source?.ip != null && ctx?.destination?.ip != null"
```
Comment (andrewkroh) [inline packages/haproxy/data_stream/log/elasticsearch/ingest_pipeline/default.yml:160]:
```suggestion
      if: ctx.source?.ip != null && ctx.destination?.ip != null
```

File: packages/haproxy/data_stream/log/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -157,11 +157,11 @@ processors:
...
   - append:
       field: event.type
       value: connection
-      if: "ctx?.source.ip != null && ctx?.destination?.ip != null"
+      if: "ctx?.source?.ip != null && ctx?.destination?.ip != null"
```
Comment (andrewkroh) [inline packages/haproxy/data_stream/log/elasticsearch/ingest_pipeline/default.yml:164]:
```suggestion
      if: ctx.source?.ip != null && ctx.destination?.ip != null
```

`ctx` can never be null in this context so we can remove the null-safe operators.

**Review by andrewkroh — COMMENTED** [review]
LGTM. Just a minor cleanup suggestion.

**Review by andrewkroh — APPROVED** [review]

File: packages/haproxy/data_stream/log/elasticsearch/ingest_pipeline/default.yml [hunk 3]
```diff
@@ -157,11 +157,11 @@ processors:
   - append:
       field: event.category
       value: network
-      if: "ctx?.source.ip != null && ctx?.destination?.ip != null"
+      if: ctx.source?.ip != null && ctx.destination?.ip != null
```
Comment (ishleenk17) [inline packages/haproxy/data_stream/log/elasticsearch/ingest_pipeline/default.yml:160]:
```suggestion
      if: ctx?.source?.ip != null && ctx?.destination?.ip != null
```

Slightly more fail safe. Adding null check for both ctx and source/destination

File: packages/haproxy/data_stream/log/elasticsearch/ingest_pipeline/default.yml [hunk 4]
```diff
@@ -157,11 +157,11 @@ processors:
...
   - append:
       field: event.type
       value: connection
-      if: "ctx?.source.ip != null && ctx?.destination?.ip != null"
+      if: ctx.source?.ip != null && ctx.destination?.ip != null
```
Comment (ishleenk17) [inline packages/haproxy/data_stream/log/elasticsearch/ingest_pipeline/default.yml:164]:
```suggestion
      if: ctx?.source?.ip != null && ctx?.destination?.ip != null
```
Same as above.

**Review by ishleenk17 — APPROVED** [review]
Please take care of the comment!
Otherwise looks good

File: packages/haproxy/data_stream/log/elasticsearch/ingest_pipeline/default.yml (hunk 3 above)
Comment (andrewkroh) [reply packages/haproxy/data_stream/log/elasticsearch/ingest_pipeline/default.yml]: @ishleenk17 `ctx` can never be null in this context, therefore it is safe to remove the null-safe operators.

This removal was done at my ask in https://github.com/elastic/integrations/pull/11594#discussion_r1823237862

**Pull request closed without merging**